### PR TITLE
[CDAP-15250] Fix the way extract plugin ids from endpoints (condition, alert/error, splitter) 

### PR DESCRIPTION
**Pull request by ajainarayanan**
**Issue**
- Plugins with "space" in their name were causing issue while building in studio.
- Primary reason for this is, while developing DAG we encode the plugin id and plugin type in css class names.
- While extracting it back we use dom's `classList` API which breaks when plugin names have spaces.

**Solution**
- Encode plugin id and names as `data-` attributes in endpoints.
- This will avoid having any custom decode logic to get node id and type

_Note_
- Have added startup script for cypress to copy all the files required for pipelines (source)
- Have also fixed other e2e tests so that we can now enable integration tests in bamboo.

JIRA: https://issues.cask.co/browse/CDAP-15250
Build: https://builds.cask.co/browse/CDAP-URUT159

### CODE REVIEW

File: cdap-ui/app/directives/dag-plus/my-dag-ctrl.js [hunk 1]
```diff
@@ -538,75 +532,27 @@ angular.module(PKG.name + '.commons')
...
-        // So this split just skips the "endpoint" and assigns the rest as nodeid.
-        let sourceIdSplit = detachedConnObj.sourceId.split('_');
-        let endIndex = sourceIdSplit.length;
-        if (detachedConnObj.sourceId.endsWith('_alert') || detachedConnObj.sourceId.endsWith('_error')) {
-          endIndex = sourceIdSplit.length - 1;
-        }
-
-        connObj.sourceId = sourceIdSplit.slice(1, endIndex).join('_');
+      if (!detachedConnObj.source || typeof detachedConnObj.source !== 'object') {
+        console.log('Detached connection is a id: ', detachedConnObj);
```
Comment (elfenheart) [inline cdap-ui/app/directives/dag-plus/my-dag-ctrl.js]: intentional?

File: cdap-ui/cypress/fixtures/null_splitter_pipeline-cdap-data-pipeline.json [hunk 2]
```diff
@@ -0,0 +1,235 @@
+{
+    "artifact": {
+        "name": "cdap-data-pipeline",
+        "version": "6.1.0-SNAPSHOT",
```
Comment (elfenheart) [inline cdap-ui/cypress/fixtures/null_splitter_pipeline-cdap-data-pipeline.json]: this should be 6.0.0

File: cdap-ui/cypress/fixtures/null_splitter_pipeline-cdap-data-pipeline.json [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        "stages": [
+            {
+                "name": "File",
+                "plugin": {
+                    "name": "File",
+                    "type": "batchsource",
+                    "label": "File",
+                    "artifact": {
+                        "name": "core-plugins",
+                        "version": "2.3.0-SNAPSHOT",
```
Comment (elfenheart) [inline cdap-ui/cypress/fixtures/null_splitter_pipeline-cdap-data-pipeline.json]: I think you'll need to build and generate these pipelines from release/6.0

File: cdap-ui/cypress/fixtures/pipeline_old.json [hunk 4]
```diff
@@ -64,7 +64,7 @@
           "type": "batchsink",
           "label": "Amazon S32",
           "artifact": {
-            "name": "amazon-s3-plugins",
+            "name": "amazon-s3-plugins1",
```
Comment (elfenheart) [inline cdap-ui/cypress/fixtures/pipeline_old.json:67]: is this change still needed?

File: cdap-ui/app/directives/dag-plus/my-dag-ctrl.js (hunk 1 above)
Comment (ajainarayanan) [reply cdap-ui/app/directives/dag-plus/my-dag-ctrl.js]: removed

File: cdap-ui/cypress/fixtures/pipeline_old.json (hunk 4 above)
Comment (ajainarayanan) [reply cdap-ui/cypress/fixtures/pipeline_old.json:67]: Talked to @elfenheart . This is there in place to make sure we have the "Missing plugins" appear in import pipeline modal which the test requires.

File: cdap-ui/cypress/fixtures/null_splitter_pipeline-cdap-data-pipeline.json (hunk 2 above)
Comment (ajainarayanan) [reply cdap-ui/cypress/fixtures/null_splitter_pipeline-cdap-data-pipeline.json]: Apparently we don't have a strict policy on cdap version from pom.xml. Some release branches seem to have version with snapshot [release/5.2 pom.xml](https://github.com/cdapio/cdap/blob/release/5.2/pom.xml#L25) and some without it [release/5.1 pom.xml](https://github.com/cdapio/cdap/blob/release/5.2/pom.xml#L25). Will change this once we change the pom.xml. (we do may be once we cut the release and push it to maven central? 🤔 Will check)

File: cdap-ui/cypress/fixtures/null_splitter_pipeline-cdap-data-pipeline.json [hunk 5]
```diff
@@ -232,4 +207,4 @@
         "numOfRecordsPreview": 100,
         "maxConcurrentRuns": 1
     }
-}
+}
```
Comment (elfenheart) [inline cdap-ui/cypress/fixtures/null_splitter_pipeline-cdap-data-pipeline.json]: nit: newline at EOF

**Review by elfenheart — APPROVED** [review]
one nit, otherwise 👍 LGTM

**Comment by ajainarayanan** [comment]
Latest green build: https://builds.cask.co/browse/CDAP-URUT159-4
